consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -278,11 +281,15 @@ impl<T: Payload> BlockStore<T> {
     /// Different execution ids are treated as different blocks (e.g., if some proposal is
     /// executed in a non-deterministic fashion due to a bug, then the votes for execution result
     /// A and the votes for execution result B are aggregated separately).
-    pub fn insert_vote(&self, vote_msg: VoteMsg, min_votes_for_qc: usize) -> VoteReceptionResult {
+    pub fn insert_vote(
+        &self,
+        vote_msg: VoteMsg,
+        validator_verifier: Arc<ValidatorVerifier>,
```

Comment:
Perhaps BlockTree might be a better location than BlockStore to avoid continually passing it down?  This will take some work, so I work prefer to make this happen in a subsequent PR.